app/src/pages/inside/common/launchFiltersContainer/launchFiltersContainer.jsx

```diff
@@ -81,28 +86,37 @@ export class LaunchFiltersContainer extends Component {
       }, {});
 
   handleFilterChange = (conditions) => {
-    this.fetchLaunches(this.createFilterQuery(conditions));
-    this.updateFilter(
-      this.props.activeFilterId,
-      Object.keys(conditions).map((key) => ({ ...conditions[key] })),
-    );
+    const newFilter = this.createFilterQuery(conditions);
+    const currentFilter = this.createFilterQuery(this.getConditions());
+
+    if (!isEqual(currentFilter, newFilter)) {
+      this.fetchLaunches(newFilter);
+    }
+
+    if (this.props.activeFilter) {
+      this.updateFilter(
+        this.props.activeFilterId,
+        Object.keys(conditions).map((key) => ({ ...conditions[key] })),
+      );
+    } else {
+      this.props.createFilter();
+    }
   };
 
   handleFilterRemove = (filter) => {
     if (filter.id >= 0) {
       this.props.hideFilterOnLaunchesAction(filter);
     }
     this.props.removeLaunchesFilterAction(filter.id);
+    this.fetchLaunches()();
```

Comment:
You're calling fetch and immediately canceling it? Looks like this line does nothing.
Also, is it necessary to update launches list here? It is already triggered when we remove a filter.